### PR TITLE
Added tutorials for Holistic Trace Analysis

### PR DESCRIPTION
**Pull request by anupambhatnagar**
Fixes [2724](https://github.com/pytorch/tutorials/issues/2724)

## Description
- Added two tutorials for HTA
  - Introduction to HTA
  - TraceDiff using HTA
- Created a left panel entry called "Profiling PyTorch" which now contains the introductory `beginner/profiler` and the two HTA tutorials. The left panel entry will allow users to find tutorials related to profiling easily. See attached image.

Screenshot 2023-12-28 at 2 01 00 PM

## Checklist

- [x] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [x] Only one issue is addressed in this pull request
- [x] Labels from the issue that this PR is fixing are added to this pull request
- [x] No unnecessary issues are included into this pull request.

### CODE REVIEW

File: requirements.txt
Comment (aaronenyeshi) [inline requirements.txt:1]: this looks like a temp change not meant to land

**Review by aaronenyeshi — CHANGES_REQUESTED** [review]

**Review by aaronenyeshi — COMMENTED** [review]
Overall looks great!

File: requirements.txt
Comment (anupambhatnagar) [reply requirements.txt:1]: yup, fixing it.

**Review by malfet — APPROVED** [review]
LGTM, not sure if `beginner` folder is the right one for it, as other profiling tutorials are in the intermediate, aren't they?

**Comment by anupambhatnagar** [comment]
> LGTM, not sure if `beginner` folder is the right one for it, as other profiling tutorials are in the intermediate, aren't they?

There are two tutorials tagged with profiling. One of them is in beginner and the other one is in intermediate. These are introductory HTA tutorials so I placed them in the beginners folder.

**Pull request merged**